lib/shoulda/matchers/active_record/validate_uniqueness_of_matcher.rb

```diff
@@ -520,8 +520,8 @@ def update_existing_record!(value)
         def arbitrary_non_blank_value
           non_blank_value = dummy_value_for(@attribute)
           limit = column_limit_for(@attribute)
-
-          if limit && limit < non_blank_value.length
+          if non_blank_value.is_a?(String) && limit &&
```

Comment:
One thing and I'll merge this -- what do you think about extracting this conditional to another method (or at the very least a variable)? The indentation mars the readability a little bit here.

Reply:
@mcmire Agreed, it does hurt legibility. Just amended the commit to put the first conditional in a variable.